compiler/src/iree/compiler/Codegen/SPIRV/Verifiers.cpp

```diff
@@ -249,45 +247,55 @@ LogicalResult verifySPIRVCooperativeMatrixVectorizePassPipeline(
   if (isa<linalg::BatchMatmulOp>(op)) {
     lhsShape = lhsShape.drop_front(1);
     rhsShape = rhsShape.drop_front(1);
-    firstLevelTileSizes.erase(firstLevelTileSizes.begin());
-    secondLevelTileSizes.erase(secondLevelTileSizes.begin());
-    thirdLevelTileSizes.erase(thirdLevelTileSizes.begin());
-  }
-
-  // Verify that subgroup tile sizes should be multiple of 16
-  if (secondLevelTileSizes[0] % 16 != 0 || secondLevelTileSizes[1] % 16 != 0 ||
-      thirdLevelTileSizes[2] % 16 != 0) {
-    return op->emitOpError("expected subgroup tile sizes to be multiple of 16");
-  }
-
-  // Verify the tile size divides the matmul inputs A [M x K] & B [K x N]
-  if (lhsShape[0] % firstLevelTileSizes[0] != 0 ||
-      lhsShape[1] % thirdLevelTileSizes[2] != 0) {
+    workgroupTileSizes.erase(workgroupTileSizes.begin());
+    subgroupTileSizes.erase(subgroupTileSizes.begin());
+    reductionTileSizes.erase(reductionTileSizes.begin());
+  }
+
+  // Verify that subgroup tile sizes should be multiple of cooperative matrix
```

Comment:
Actually as said before, we should first check the fourth level against the cooperative matrix properties. Something like the following:

```
SmallVector<int64_t> nativeVectorSizes = loweringConfig.getTileSizeVals(3);
...
auto getElementType = [](Value v) { return v.getType().cast<ShapedType>().getElementType(); };
Type aType = getElementType(lhs);
Type bType = getElementType(rhs);
Type cType = getElementType(result);
...
if (isa<linalg::BatchMatmulOp>(op)) {
  ...
  nativeVectorSizes.erase(nativeVectorSizes.begin());
}
...

bool isNativeVectorSizeAccepted = false;
for (auto p : properties) {
  if (p.getAType() == aType && p.getBType() == bType && p.getCType == cType() &&
      p.getScope().getValue() == spirv::Scope::Subgroup &&
      p.getMSize() == nativeVectorSizes[0] && p.getNSize() == nativeVectorSizes[1] && p.getKSize() == nativeVectorSizes[2])
  isNativeVectorSizeAccepted = true;
}
if (!isNativeVectorSizeAccepted) op->emitOpError(...);

// Then go on to check subgroup tile sizes is a multiple of nativeVectorSizes
```

Reply:
Thanks for pointing out the problem. I've made the changes.